components/cost-management/policies/update_namespace_labels.yaml

```
@@ -0,0 +1,72 @@
+---
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: propagate-cost-center-from-namespace
+  annotations:
+    policies.kyverno.io/title: Propagate Cost-Center from Namespace
+    policies.kyverno.io/category: Cost Management
+    policies.kyverno.io/severity: low
+    policies.kyverno.io/subject: Pod, Namespace
+    policies.kyverno.io/description: >-
+      This policy ensures that the cost-center label from a Namespace is propagated
+      to all Pods within that Namespace. It applies only to Namespaces with the
+      konflux-ci.dev/type: tenant label and ensures that the cost-center label exists
+      in the Namespace before applying the mutation.
+spec:
+  rules:
+    - name: propagate-missing-cost-center-from-namespace
+      match:
+        any:
+        - resources:
+            kinds:
+            - Pod
+            namespaceSelector:
+              matchLabels:
+                konflux-ci.dev/type: tenant
+      context:
+        - name: costcenterLabel
+          apiCall:
+            urlPath: "/api/v1/namespaces/{{ request.object.metadata.namespace }}"
+            method: GET
+            jmesPath: 'metadata.labels."cost-center" || ""'
+      preconditions:
+        all:
+          - key: "{{ costcenterLabel }}"
+            operator: NotEquals
+            value: ""
+      mutate:
+        targets:
+          - apiVersion: v1
+            kind: Pod
+        patchStrategicMerge:
+          metadata:
+            labels:
+              cost-center: "760"
+              insights_cost_management_optimizations: "true"
```

I'd suggest using hyphens instead of underscores here:

```suggestion
              cost-center: "760"
              insights-cost-management-optimizations: "true"
```

---

I think this is going to be a problem for Kyverno as it only accepts the following pattern ->^(?:\w+|\(.+\))$
(it doesn't accept -)